File: src/main/java/com/everyonewaiter/application/auth/service/request/SendAuthCode.java
package com.everyonewaiter.application.auth.service.request;

import com.everyonewaiter.domain.auth.entity.AuthPurpose;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class SendAuthCode {

  private String phoneNumber;
  private AuthPurpose purpose;

}

File: src/main/kotlin/com/everyonewaiter/application/auth/dto/SendAuthCode.kt
package com.everyonewaiter.application.auth.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.Pattern

class SendAuthCode {
    @Schema(name = "SendAuthCode.Request")
    data class Request(
        @Schema(description = "휴대폰 번호", example = "01044591812")
        @field:Pattern(regexp = "^01[016789]\\d{7,8}$", message = "잘못된 형식의 휴대폰 번호를 입력하셨어요. 휴대폰 번호를 다시 입력해 주세요.")
        val phoneNumber: String,
    )
}
